workers/unity/Assets/Fps/Scripts/GameLogic/StateMachine/PlayState.cs

```diff
@@ -0,0 +1,145 @@
+        {
+            this.connector = connector;
+            trackPlayerSystem = connector.Worker.World.GetExistingManager<TrackPlayerSystem>();
+            lobbyScreenController = controller.FrontEndController.LobbyScreenController;
+        }
+
+        public override void StartState()
+        {
+            Controller.FrontEndController.LobbyScreenController.playerNameInputController.OnNameChanged += OnNameUpdated;
+            lobbyScreenController.startButton.onClick.AddListener(StartSession);
+            lobbyScreenController.cancelButton.onClick.AddListener(CancelSession);
+            lobbyScreenController.ConnectionStatusUIController.ShowWaitForGameText();
+            queryDeploymentStatus = true;
+        }
+
+        public override void ExitState()
+        {
+            lobbyScreenController.startButton.onClick.RemoveListener(StartSession);
+            lobbyScreenController.cancelButton.onClick.RemoveListener(CancelSession);
+        }
+
+        public override void Tick()
+        {
+            time += Time.deltaTime;
+
+            if (!queryDeploymentStatus || time < 1f)
+            {
+                return;
+            }
+
+            time = 0;
+
+            Status? status;
+            if (hasGameStarted)
+            {
+                status = trackPlayerSystem.SessionStatus;
+            }
+            else
+            {
+                status = GetStatus();
+                if (status == Status.RUNNING)
+                {
+                    lobbyScreenController.startButton.enabled = status == Status.RUNNING && hasValidName;
+                    lobbyScreenController.ConnectionStatusUIController.ShowGameReadyText();
+                }
+            }
+
+            if (status == Status.STOPPING)
+            {
+                Owner.SetState(new ResultsState(connector, Controller, Owner));
+            }
+        }
+
+        private void OnNameUpdated(bool isValid)
+        {
+            hasValidName = isValid;
+        }
+
+        private void StartSession()
+        {
+            queryDeploymentStatus = false;
+            lobbyScreenController.startButton.enabled = false;
+            connector.SpawnPlayerAction(Controller.FrontEndController.LobbyScreenController.GetPlayerName(), CreatedPlayer);
+            lobbyScreenController.ConnectionStatusUIController.ShowSpawningText();
+        }
+
+        private void CreatedPlayer(PlayerCreator.CreatePlayer.ReceivedResponse obj)
+        {
+            if (obj.StatusCode == StatusCode.Success)
+            {
+                Controller.ShowGameView();
+                hasGameStarted = true;
+                queryDeploymentStatus = true;
+            }
+            else
+            {
+                UnityEngine.Debug.LogWarning(obj.StatusCode);
+                UnityObjectDestroyer.Destroy(connector.gameObject);
```

Comment:
Can this run in editor mode? of not, no need to use `UnityObjectDestroyer`.

Reply:
it can run in editor mode